theano/gpuarray/opt.py

```diff
@@ -183,7 +183,11 @@ def f(local_opt):
 # in order to avoid introducin new CPU Ops, or useless ones.
 def safe_to_gpu(x, ctx_name):
     if isinstance(x.type, tensor.TensorType):
-        return GpuFromHost(ctx_name)(x)
+        ret = GpuFromHost(ctx_name)(x)
+        values_eq_approx = getattr(x.tag, 'values_eq_approx', None)
+        if values_eq_approx:
+            ret.tag.values_eq_approx = values_eq_approx
```

Comment:
I think this is good, but maybe we might want to do the tag transfer in a more general level. Like in the transfer functions such as `GpuFromHost` or `transfer`.

Reply:
I did it. Good idea.